in_toto/keylib.go

```diff
@@ -52,10 +52,14 @@ file at path does not exist or is not a PEM formatted RSA public key.
 */
 func (k *Key) LoadPublicKey(path string) error {
 	keyFile, err := os.Open(path)
-	defer keyFile.Close()
 	if err != nil {
 		return err
 	}
+	defer func() {
```

Comment:
My problem right now is in error handling like this. For me it makes sense to at least inform the  user about an error here, but we are unable to properly test error handling for it. At least I don't know how to provoke a keyFile.Close() error

Reply:
That's fine, we really don't have to emulate the error on close. If necessary you can reduce the required coverage in the test suite.

Also, is logging the error the right approach here? Wouldn't it be better to wrap the close error into the error returned by LoadPublicKey?

Reply:
Mh yes, makes sense. but how would I wrap this into the error returned by the LoadPublicKey() func? Is `defer` passing the error through? I mean If I would do `return err` inside of this anonymous function, would it get passed?

Reply:
Looks like the `errors` package allows you to do such things. Check out [this article](https://blog.learngoprogramming.com/5-gotchas-of-defer-in-go-golang-part-iii-36a1ab3d6ef1) and search for *"You can also use this package to wrap multiple errors"*

Reply:
Ok I can just assign to the err variable from the upper function, like this:
https://play.golang.org/p/BeGOyTH7yPE

I will do this :)

Reply:
Done